packages/xstate-react/src/useMachine.ts

```diff
@@ -142,18 +148,22 @@ export function useMachine<
       services,
       delays
     };
-
-    const createdMachine = machine.withConfig(machineConfig, {
+    const resolvedMachine = machine.withConfig(machineConfig, {
       ...machine.context,
       ...context
     } as TContext);
 
-    return interpret(createdMachine, interpreterOptions).start(
-      rehydratedState ? State.create(rehydratedState) : undefined
-    );
+    return [
+      resolvedMachine,
+      interpret(resolvedMachine, { deferEvents: true, ...interpreterOptions })
```

Comment:
I **think** (not 100% sure) that this change has actually fixed the test mentioned here:
https://github.com/davidkpiano/xstate/pull/1239#discussion_r445583363

Would be great to check if it would indeed fail without this and add it if it does (still might be worth adding regardless as it's a slight variation over the existing test, but that would be up to you)

Reply:
Test was added!